### ISSUE TITLE
[Question][Hw1][Problem 4] Ambiguity in exception

### ISSUE THREAD

**Issue by ShinWonho** [prosyslab-classroom/cs524-program-analysis#12]
Hi, I think there is some ambiguity in raising an exception.
`Add (Div (Int 1, Int 0), X)`
In this case, it is ambiguous to raise `Invalid_argument` or `Division_by_zero` because no specific evaluation order is given.
In those ambiguous cases, is there any specific order of evaluation or any exception is permitted?

**Comment by KihongHeo** [comment]
Good point. I am happy to see that you clearly understand undefined behavior in PL semantics.  
As usual compilers and interpreters, it depends on your choice :).
(i.e., TAs will not use such an ambiguous example for evaluation).

**Comment by ShinWonho** [comment]
Thanks you professor!
